File: src/main/java/de/spries/fleetcommander/persistence/GameStore.java
package de.spries.fleetcommander.persistence;

import java.util.Collection;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

import de.spries.fleetcommander.model.core.Game;

public enum GameStore {
	INSTANCE;

	private Map<Integer, Game> gameStore;
	private int nextGameId;

	private GameStore() {
		reset();
	}

	public synchronized int create(Game game) {
		gameStore.put(nextGameId, game);
		return nextGameId++;
	}

	public Game get(int id) {
		return gameStore.get(id);
	}

	public void delete(int id) {
		gameStore.remove(id);
	}

	protected Collection<Game> getGames() {
		return gameStore.values();
	}

	protected void reset() {
		gameStore = new ConcurrentHashMap<>();
		nextGameId = 1;
	}
}

File: src/main/kotlin/de/spries/fleetcommander/persistence/GameStore.kt
package de.spries.fleetcommander.persistence

import de.spries.fleetcommander.model.core.Game
import java.util.concurrent.ConcurrentHashMap

enum class GameStore {
    INSTANCE;

    private val gameStore: MutableMap<Int, Game> = ConcurrentHashMap()
    private var nextGameId: Int = 1

    val games: Collection<Game>
        get() = gameStore.values

    @Synchronized
    fun create(game: Game): Int {
        gameStore[nextGameId] = game
        return nextGameId++
    }

    operator fun get(id: Int): Game? {
        return gameStore[id]
    }

    fun delete(id: Int) {
        gameStore.remove(id)
    }

    fun reset() {
        gameStore.clear()
        nextGameId = 1
    }
}
